src/clusterfuzz/_internal/bot/fuzzers/utils.py

```diff
@@ -72,22 +75,23 @@ def is_fuzz_target_local(file_path, file_handle=None):
     logs.warning('Tried to read from non-regular file: %s.' % file_path)
     return False
 
-  # Use already provided file handle or open the file.
-  local_file_handle = file_handle or open(file_path, 'rb')
+  # Either use the file opener or open the file ourselves.
+  if not file_opener:
+    file_opener = functools.partial(open, mode='rb')
+  file_handle = file_opener(file_path)
+  if not file_handle:
+    return False
 
   result = False
   for pattern in FUZZ_TARGET_SEARCH_BYTES:
     # TODO(metzman): Bound this call so we don't read forever if something went
     # wrong.
-    local_file_handle.seek(0)
-    result = utils.search_bytes_in_file(pattern, local_file_handle)
+    file_handle.seek(0)
+    result = utils.search_bytes_in_file(pattern, file_handle)
```

Comment:
Yes, but:
1. We ultimately only download a fuzz target twice, because the other ones won't be selected to be unpacked, thus those will only be downloaded once.
2. This is already true anyways. And yes, I agree this is bad, especially the code you're pointing to.